polkadot/xcm/src/v3/multiasset.rs

```diff
@@ -974,4 +980,31 @@ mod tests {
 		let r = MultiAssets::from_sorted_and_deduplicated(mixed_bad);
 		assert!(r.is_err());
 	}
+
+	#[test]
+	fn decoding_respects_limit() {
+		use super::*;
+
+		// Having lots of one asset will work since they are deduplicated
+		let lots_of_one_asset: MultiAssets =
+			vec![(GeneralIndex(1), 1u128).into(); MAX_ITEMS_IN_MULTIASSETS + 1].into();
+		let encoded = lots_of_one_asset.encode();
+		assert!(MultiAssets::decode(&mut &encoded[..]).is_ok());
+
+		// Fewer assets than the limit works
+		let mut few_assets: MultiAssets = Vec::new().into();
+		for i in 0..MAX_ITEMS_IN_MULTIASSETS - 1 {
```

Comment:
We want to test the edge case as much as possible, so let's do this:

```suggestion
		for i in 0..MAX_ITEMS_IN_MULTIASSETS {
```